src/VecSim/spaces/L2_space.cpp

```diff
@@ -145,6 +154,7 @@ dist_func_t<float> L2_BF16_GetDistFunc(size_t dim, const Arch_Optimization arch_
             *alignment = 16 * sizeof(bfloat16); // align to 256 bits.
         break;
 #endif
+    case ARCH_OPT_F16C:
```

Comment:
since f16c is above avx2 in spaces_aux, i won't get here the most optimized function.